### DISCUSSION TITLE
Removing support for BRG library

### DISCUSSION THREAD

**Discussion by nmoinvaz** [nmoinvaz/minizip#545] [Ideas]
I am planning on removing support for BRG library since I have native crypto implementations on the major platforms. If one of the crypto implementations is not found, it will disable encryption support. This should remove both the `MZ_BRG` and `MZ_LIBBSD` cmake options as libbsd will no longer be necessary.

I would like to move to third-party libraries that have large adoption and active maintenance.

**Comment by Nekto89** [comment]
I'm currently trying to switch from old minizip 1.10 to your version 2.10.5. Have I understood correctly, that on Linux the only way to get old MZ_PKCRYPT working (without AES) is to use BRG or OpenSSL? So with removal of BRG there will be only one option - OpenSSL?

**Reply by nmoinvaz** [reply]
Yes for Linux. Windows and macOS have their own native crypto libraries. We can add support for another popular crypto library if necessary.

**Reply by gvollant** [reply]
Is there a specific problem with BRG?

**Reply by nmoinvaz** [reply]
It requires two separate repositories (1 for AES and 1 for SHA1) and it also requires libbsd for cryptographic random function. I would like to simplify by using libraries that contain all those such as openSSL or mbedTLS.

**Reply by nmoinvaz** [reply]
I have made it so it is not necessary to use OpenSSL to get MZ_PKCRYPT support.
